netlogo-gui/src/main/app/NotificationBanner.scala

```diff
@@ -0,0 +1,218 @@
+  initUI()
+
+  override def paintComponent(g: Graphics) {
+    if (isHover)
+      setBackground(InterfaceColors.ANNOUNCEMENTS_BANNER_BACKGROUND_HOVER)
+    else
+      setBackground(InterfaceColors.ANNOUNCEMENTS_BANNER_BACKGROUND)
+
+    super.paintComponent(g)
+  }
+  def syncTheme(): Unit = {
+    setBackground(InterfaceColors.ANNOUNCEMENTS_BANNER_BACKGROUND)
+    messageLabel.setForeground(InterfaceColors.ANNOUNCEMENTS_BANNER_TEXT)
+    closeButton.setForeground(InterfaceColors.ANNOUNCEMENTS_BANNER_TEXT)
+  }
+
+  // Method to fetch JSON content from a URL
+  private def fetchJsonFromUrl(): String = {
+    try {
+      val source = Source.fromURL(jsonUrl)
+      val content = source.mkString
+      source.close()
+      content
+    } catch {
+      case e: Exception =>
+      println(s"Unable to connect to ccl.northwestern.edu but not throwing an exception")
+      ""
+    }
+  }
+
+  private def initUI(): Unit = {
+
+    // Set layout and appearance
+    setLayout(new BorderLayout())
+    setPreferredSize(new Dimension(400, 50))
+
+    // Customize the message label
+    messageLabel.setHorizontalAlignment(SwingConstants.LEFT)
+    add(messageLabel, BorderLayout.CENTER)
+
+    // Configure the close button
+    closeButton.setBorderPainted(false)
+    closeButton.setContentAreaFilled(false)
+    closeButton.setFocusPainted(false)
+    //TODO this action needs to get called on the okay button click.
+    closeButton.addActionListener(new ActionListener {
+      override def actionPerformed(e: ActionEvent): Unit = {
+        setVisible(false) // Hide the banner when the close button is pressed
+      }
+    })
+    add(closeButton, BorderLayout.EAST)
+
+    // Add a mouse listener to call showJsonInDialog when the banner is clicked
+    addMouseListener(new MouseAdapter {
+      override def mouseClicked(e: MouseEvent): Unit = {
+        showJsonInDialog() // Call the method on click
+      }
+    })
+  }
+
+  // Method to parse JSON content to a list of JsonObject instances
+  private def parseJsonToList(jsonContent: String): List[JsonObject] = {
+    if(jsonContent.isEmpty){
+      return Nil
+    }
+
+      val parser = new JSONParser()
+      val jsonArray = parser.parse(jsonContent).asInstanceOf[JSONArray]
+      jsonArray.toArray.flatMap { obj =>
+        val jsonObject = obj.asInstanceOf[JSONObject]
+        Option(jsonObject.get("event-id").asInstanceOf[Long].toInt).map { eventId =>
+          val title = Option(jsonObject.get("title")).map(_.toString).getOrElse("")
+          val fullText = Option(jsonObject.get("fullText")).map(_.toString).getOrElse("")
+          val date = Option(jsonObject.get("date")).map(_.toString).getOrElse("")
+          JsonObject(eventId, date, title, fullText)
+        }
+      }.toList.sortBy(_.eventId)(Ordering[Int].reverse)
+
+  }
+
+  // Method to show JSON content in a dialog
+  private def showJsonInDialog(): Unit = {
+
+    val prefs = Preferences.userNodeForPackage(getClass)
```

Comment:
Done

Reply:
This is still not done, but you've pushed multiple commits since then. To help yourself and us, please don't mark it as done if you haven't pushed the change already or are about to, just in case you lose track and forget to implement it.